specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json

```diff
@@ -2853,6 +3049,154 @@
                 }
             ]
         },
+        "NorthSouthProtocol": {
+            "description": "The transport protocols supported by north-south hardenings",
+            "type": "string",
+            "enum": [
+                "TCP",
+                "UDP"
+            ],
+            "x-ms-enum": {
+                "name": "northSouthProtocol",
+                "modelAsString": true
+            }
+        },
+        "NorthSouthTrafficDirection": {
+            "description": "The north south traffic direction",
+            "type": "string",
+            "enum": [
+                "Inbound",
+                "Outbound"
+            ],
+            "x-ms-enum": {
+                "name": "northSouthTrafficDirection",
+                "modelAsString": true
+            }
+        },
+        "TrafficHardeningRule": {
+            "description": "A north-south traffic hardening rule",
+            "properties": {
+                "name": {
+                    "description": "The name of the rule",
+                    "type": "string"
+                },
+                "direction": {
+                    "description": "The rule;s traffic direction",
+                    "$ref": "#/definitions/NorthSouthTrafficDirection"
+                },
+                "destinationPort": {
+                    "description": "The rule's destination port",
+                    "$ref": "#/definitions/PortNumber"
+                },
+                "protocols": {
+                    "description": "The rule's transport protocol",
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/NorthSouthProtocol"
+                    }
+                },
+                "allowRemoteAddresses": {

```

Comment:
does 'allow**ed**RemoteAddresses' make more sense than 'allowRemoteAddresses'

Reply:
allowed (past tense) hints that the trafficHardeningRules already applies on the VM's NSG while these set of rules hasn't applied yet on the NSG (some/all of the rules may be enforced on the NSG using the Enforce action).